tests/Feature/Platform/TestsPlayerAchievements.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Platform;
+
+use Carbon\Carbon;
+use LegacyApp\Platform\Enums\UnlockMode;
+use LegacyApp\Platform\Models\Achievement;
+use LegacyApp\Platform\Models\PlayerAchievement;
+use LegacyApp\Site\Models\User;
+
+trait TestsPlayerAchievements
+{
+    protected function addPlayerAchievement(User $user, Achievement $achievement,
+        ?Carbon $hardcoreUnlockTime, Carbon $softcoreUnlockTime): void
+    {
```

Comment:
```suggestion
    protected function addPlayerAchievement(
        User $user, 
        Achievement $achievement,
        ?Carbon $hardcoreUnlockTime, 
        Carbon $softcoreUnlockTime
    ): void {
```

Personally I prefer to go all with chopping down parameters when introducing new lines - not very important though.